definitions/checks/sca_only.rb

```diff
@@ -0,0 +1,42 @@
+class Checks::CheckOrganizationContentAccessMode < ForemanMaintain::Check
+  metadata do
+    label :check_organization_content_access_mode
+    description 'Check if any organizations are using entitlement mode before upgrading'
+    tags :pre_upgrade
+
+    confine do
+      feature(:katello)
+    end
+  end
+
+  def run
+    with_spinner('Checking organization content access modes') do
+      output = execute_rake_task
+      orgs_to_migrate = parse_orgs_to_migrate(output)
+
+      assert(
+        orgs_to_migrate.empty?,
+        "The following organizations are using entitlement mode:\n"\
+        "#{orgs_to_migrate.join(', ')}\n"\
+        "As of Katello 4.12, entitlement mode is removed, and these organizations"\
+        " will be automatically migrated to Simple Content Access during the upgrade."\
+        " Please ensure that you have reviewed the documentation and are prepared for this change."\
+      )
+    end
+  end
+
+  private
+
+  def execute_rake_task
+    execute!('foreman-rake katello:upgrades:4.12:update_content_access_modes[dry_run]')
```

Comment:
If we're upgrading from Katello 4.11, and this rake task is only included in 4.12, will we have access to it here?

Reply:
If this check is intended to run before upgrading then no, as the Katello code will only be updated during the upgrade steps after pre upgrade checks.

Reply:
Hm, we'll need a different solution here then.

Can the pre-upgrade check run a `rails runner` command?

```
foreman-rake runner "Organization.all.map(&:simple_content_access?).count(false)"
```

should give you the number of orgs you need to compose the message here, without needing to backport the rake task.

Reply:
I'm spinning up a katello 4.11 to test this

Reply:
trying to figure out why this is happening:

```
[root@centos8-stream-katello-4-11 ~]# foreman-maintain upgrade list-versions --disable-self-upgrade
[root@centos8-stream-katello-4-11 ~]# 
```